src/apps/testapps/testEdge.c

```diff
@@ -0,0 +1,298 @@
+/*
+ * Copyright 2022 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/** @file testEdge.c
+ * @brief Tests functions for manipulating undirected edge H3Indexes
+ *
+ * usage: `testEdge`
+ */
+
+#include <stdlib.h>
+
+#include "algos.h"
+#include "constants.h"
+#include "h3Index.h"
+#include "latLng.h"
+#include "test.h"
+#include "utility.h"
+
+// Fixtures
+static LatLng sfGeo = {0.659966917655, -2.1364398519396};
+
+SUITE(edge) {
+    TEST(cellsToEdgeAndFriends) {
+        H3Index sf;
+        t_assertSuccess(H3_EXPORT(latLngToCell)(&sfGeo, 9, &sf));
+        H3Index ring[7] = {H3_NULL};
+        t_assertSuccess(H3_EXPORT(gridRingUnsafe)(sf, 1, ring));
+        H3Index sf2 = ring[0];
+
+        H3Index edge;
+        t_assertSuccess(H3_EXPORT(cellsToEdge)(sf, sf2, &edge));
+        H3Index cells[2] = {H3_NULL};
+        t_assertSuccess(H3_EXPORT(edgeToCells)(edge, cells));
+        t_assert(cells[0] == sf || cells[1] == sf,
+                 "One of the cells is the origin");
+        t_assert(cells[0] == sf2 || cells[1] == sf2,
+                 "One of the cells is the destination");
+        t_assert(sf != sf2, "Sanity check for cells not being the same");
+        t_assert(cells[0] < cells[1],
+                 "Directed edge cells are in normalized order");
```

Comment:
Nit: Not sure it makes sense to test this, as from an API perspective the order of these two cells is undefined. Also the comment doesn't make sense (should be "undirected", but then the order doesn't matter)

Reply:
This is relying on an implementation detail that the first cell returned is the "origin". We could choose to make that a defined part of the API.

I'll fix the comment.